src/parse_fc.c

```diff
@@ -243,22 +251,52 @@ struct policy_node *parse_fc_file(const char *filename, const struct string_list
 	struct policy_node *cur = head;
 
 	char *line = NULL;
+	char *ifdef_condition = NULL;
+	char *ifndef_condition = NULL;
+	char *token = NULL;
+	struct conditional_data *conditional = NULL;
 
 	ssize_t len_read = 0;
 	size_t buf_len = 0;
 	unsigned int lineno = 0;
+	bool is_within_ifdef = false;
+	bool is_within_ifndef = false;
 	while ((len_read = getline(&line, &buf_len, fd)) != -1) {
 		lineno++;
 		if (len_read <= 1 || line[0] == '#') {
 			continue;
-		}
-		// Skip over m4 constructs
-		if (strncmp(line, "ifdef", 5) == 0 ||
-		    strncmp(line, "ifndef", 6) == 0 ||
-		    strncmp(line, "')", 2) == 0 ||
-		    strncmp(line, "', `", 4) == 0 ||
-		    strncmp(line, "',`", 3) == 0) {
-
+		} else if (!strncmp(line, "ifdef", 5)) {
+			is_within_ifdef = true;
+			token = strtok(line, "`");
+			if (token) {
+				token = strtok(NULL, "`");
+				token = strtok(token, "'");
+				ifdef_condition = malloc(strlen(token) + 1);
+				ifdef_condition = strdup(token);
+			}
+			continue;
+		} else if (!strncmp(line, "ifndef", 6)) {
+			is_within_ifndef = true;
+			token = strtok(line, "`");
+			if (token) {
+				token = strtok(NULL, "`");
+				token = strtok(token, "'");
+				ifndef_condition = malloc(strlen(token) + 1);
+				ifndef_condition = strdup(token);
+			}
+			continue;
+		} else if (!strncmp(line, "')", 2)) {
+			if (is_within_ifdef) {
+				is_within_ifdef = false;
+				free(ifdef_condition);
+				ifdef_condition = NULL;
+			} else if (is_within_ifndef) {
+				is_within_ifndef = false;
+				free(ifndef_condition);
+				ifndef_condition = NULL;
+			}
+			continue;
+		} else if (!strncmp(line, "', `", 4) || !strncmp(line, "',`", 3)) { // Skip over m4 constructs
```

Comment:
This looks broken.  This would typically flip the ifdef.  These are for situations like:

```
ifdef(`foo',`
/some/path label1
',`
/some/path label2
')

In order to assign label1 if foo was set and label2 if it's not.  If I'm reading this right, this would treat both as being in the ifdef and result in a false positive in this scenario.

Reply:
that is correct. I failed to take this situation into account. so I'll need to rework this